docs/filesystem.rst

```diff
@@ -0,0 +1,179 @@
+Local Persistent File System
+****************************
+
+It is useful to store data in a persistent manner so that it remains intact
+between restarts of the device. On traditional computers this is often achieved
+by a file system consisting of named files that hold raw data, and named
+directories that contain files. Python supports the various operations needed
+to work with such file systems.
+
+However, since the micro:bit is a limited device in terms of both hardware and
+storage capacity MicroPython provides a small subset of the functions needed
+to persist data on the device. Because of memory constraints **there is
+approximately 30k of storage available** on the filesystem.
+
+.. warning::
+
+    Re-flashing the device will DESTROY YOUR DATA.
+
+    Since the file system is stored in the micro:bit's flash memory and
+    flashing the device rewrites all the available flash memory then all your
+    data will be lost if you flash your device.
+
+    However, if you switch your device off the data will remain intact until
+    you either delete it (see below) or re-flash the device.
+
+MicroPython provides a flat file system; i.e. there is no notion of a directory
```

Comment:
Can we reword this so it doesn't sound like all uPy implementations provide just a flat FS?  Eg something like "MicroPython on the micro:bit"?